src/Umbraco.Core/Persistence/Repositories/VersionableRepositoryBase.cs

```diff
@@ -238,24 +238,54 @@ private Sql GetFilteredSqlForPagedResults(Sql sql, Func<Tuple<string, object[]>>
             return filteredSql;
         }
 
-        private Sql GetSortedSqlForPagedResults(Sql sql, Direction orderDirection, string orderBy)
+        private Sql GetSortedSqlForPagedResults(Sql sql, Direction orderDirection, string orderBy, bool orderBySystemField)
         {
             //copy to var so that the original isn't changed
             var sortedSql = new Sql(sql.SQL, sql.Arguments);
-            // Apply order according to parameters
-            if (string.IsNullOrEmpty(orderBy) == false)
+
+            if (orderBySystemField)
             {
-                var orderByParams = new[] { GetDatabaseFieldNameForOrderBy(orderBy) };
-                if (orderDirection == Direction.Ascending)
+                // Apply order according to parameters
+                if (string.IsNullOrEmpty(orderBy) == false)
                 {
-                    sortedSql.OrderBy(orderByParams);
+                    var orderByParams = new[] { GetDatabaseFieldNameForOrderBy(orderBy) };
+                    if (orderDirection == Direction.Ascending)
+                    {
+                        sortedSql.OrderBy(orderByParams);
+                    }
+                    else
+                    {
+                        sortedSql.OrderByDescending(orderByParams);
+                    }
                 }
-                else
+            }
+            else 
+            { 
+                // Sorting by a custom field, so set-up sub-query for ORDER BY clause to pull through valie
+                // from most recent content version for the given order by field
+                sortedSql.Append(@"ORDER BY (
+	                SELECT CASE
+		                WHEN dataInt Is Not Null THEN RIGHT('00000000' + CAST(dataInt AS varchar(8)),8)

```

Comment:
This will not work with MySql ... the problem is IIRC 'Right' and 'Convert'

In order to do that properly you'd have to modify the ISqlSyntax object to return the correct syntax for this per data base type.

We'll prob have to postpone this after 7.3 until we can make it compatible